pkg/api/scylla/v1alpha1/types_operatorconfig.go

```diff
@@ -5,12 +5,49 @@ package v1alpha1
 import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
 type ScyllaOperatorConfigSpec struct {
-	// scyllaUtilsImage is a Scylla image used for running scylla utilities.
-	// +kubebuilder:validation:MinLength=1
+	// scyllaUtilsImage is a ScyllaDB image used for running ScyllaDB utilities.
 	ScyllaUtilsImage string `json:"scyllaUtilsImage"`
```

Comment:
shouldn't this be marked as optional now? Or does that not work without being able to switch to a pointer?

Reply:
I suppose you might convert a non optional field to a optional pointer that's omit empty but we don't need to make a change on the wire to distinguish not set from empty. This just retains the existing behaviour and loosens the validation.

Reply:
I think that if it's possible it'd be good to do it for the sake of consistency alone

Reply:
Why? The desired state is vice versa but that can't be done for the sake of compatibility. We only have the 2 state logic here.

Reply:
> > I think that if it's possible it'd be good to do it for the sake of consistency alone
> Why?

Why what? Why is it good to have consistency? It's an optional field now being indistinguishable from a required field by just looking at the API, it just doesn't have any validation

Reply:
It used to be a required field.

> It's an optional field

No, it is still a required field. It has a special value `""` which says use the same version as the operator. It is required to still be present on the wire, as it used to be, even if empty. 

>  it just doesn't have any validation

It does have validation. It only allows an empty string as a valid value.

> Why is it good to have consistency? 

You are asking to make a wire change. Old fields stay how they were and new fields are optional by design. That's how APIs evolve. In a next API version these can be reconsidered freely.

> now being indistinguishable from a required field

This is not just a hint - it changes the wire representation. Old decoders won't be able to decode the type anymore and fail on a missing the required field, if strict. There is a high bar to do wire changes. It may be necessary some days for a higher good and limiting the focus on the major decoders and our code and the kube-apiserver and accepting that the most of the system keeps working.